src/base/bittorrent/peerinfo.cpp

```diff
@@ -408,3 +408,8 @@ QString PeerInfo::flagsDescription() const
 {
     return m_flagsDescription;
 }
+
+int PeerInfo::downloadingPieceIndex() const
```

Comment:
Ok. Moreover, `TorrentInfo::filesForPiece()` is needed for `PeerInfo::downloadingFiles()` too. It just has to be so:
```c++
return m_torrent->info().filesForPiece(m_nativeInfo.downloading_piece_index);
```